File: backend/src/routes/approvals.js
const express = require('express');
const router = express.Router();

// POST /api/approvals/:id/approve
router.post('/:id/approve', async (req, res) => {
  try {
    const { resolveApproval } = require('../orchestrator/caseOrchestrator');
    const result = await resolveApproval(req.params.id, true);
    res.json(result);
  } catch (error) {
    console.error('[Approvals] Error approving:', error);
    res.status(500).json({ error: error.message });
  }
});

// POST /api/approvals/:id/reject
router.post('/:id/reject', async (req, res) => {
  try {
    const { resolveApproval } = require('../orchestrator/caseOrchestrator');
    const result = await resolveApproval(req.params.id, false);
    res.json(result);
  } catch (error) {
    console.error('[Approvals] Error rejecting:', error);
    res.status(500).json({ error: error.message });
  }
});

module.exports = router;

File: backend/app/routers/approvals.py
from fastapi import APIRouter, HTTPException
from app.orchestrator.case_orchestrator import resolve_approval

router = APIRouter(prefix="/approvals", tags=["approvals"])

@router.post("/{id}/approve")
async def approve_case(id: str):
    try:
        result = await resolve_approval(id, True)
        return result
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.post("/{id}/reject")
async def reject_case(id: str):
    try:
        result = await resolve_approval(id, False)
        return result
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
